lib/http/node.js

```diff
@@ -76,17 +76,28 @@ AWS.NodeHttpClient = AWS.util.inherit({
 
   writeBody: function writeBody(stream, httpRequest) {
     var body = httpRequest.body;
-
-    if (body && WritableStream && ReadableStream) { // progress support
-      if (!(body instanceof Stream)) body = AWS.util.buffer.toStream(body);
-      body.pipe(this.progressStream(stream, httpRequest));
-    }
+    var totalBytes = parseInt(httpRequest.headers['Content-Length'], 10);
 
     if (body instanceof Stream) {
-      body.pipe(stream);
+      // For progress support of streaming content -
+      // pipe the data through a transform stream to emit 'sendProgress' events
+      body.pipe(this.progressStream(stream, totalBytes)).pipe(stream);
```

Comment:
BTW I remembered correctly that using multiple writable streams behaves weird in nodejs v4 - I just submitted this issue https://github.com/nodejs/node/issues/6491. 
It was actually only fixed on nodejs v5.11 but I guess they will want to pull back to v4 for LTS.
Anyway that's a good reason to use Transform instead.